### PR TITLE
Include the self-hosted flag in GH Workflows

### PR DESCRIPTION
**Pull request by todd-mystenlabs**
Trying out a self-hosted GitHub  runner on Linux x86. Switching Rust to use it.  We're running out of juice on built-in runners.

### CODE REVIEW

**Comment by Andrew47** [comment]
Regarding the `Explorer Client PR checks`, the issue is specifically with Puppeteer. The troubleshooting guide suggests a dependency may be missing: https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md.

**Comment by Andrew47** [comment]
The Rust test fails also look like they are because of missing dependencies:

* error: linker `cc` not found

* Error: docker: command not found

**Comment by todd-mystenlabs** [comment]
Okay looks like we got the rest of the kinks worked out.  I'll merge this once I have runners ready to really accept traffic.

File: .github/workflows/bench.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ env:
 
 jobs:
   diff:
-    runs-on: ubuntu-latest
+    runs-on: [self-hosted, ubuntu-latest]
```
Comment (bmwill) [inline .github/workflows/bench.yml:31]: This is a cheap job so maybe don't put it on the runners?

File: .github/workflows/codecov.yml [hunk 2]
```diff
@@ -10,7 +10,7 @@ on:
 
 jobs:
   diff:
-    runs-on: ubuntu-latest
+    runs-on: [self-hosted, ubuntu-latest]
```
Comment (bmwill) [inline .github/workflows/codecov.yml:13]: This is a cheap job so maybe don't put it on the runners?

File: .github/workflows/explorer-client-prs.yml [hunk 3]
```diff
@@ -2,7 +2,7 @@ name: Explorer Client PR checks
 on: pull_request
 jobs:
   diff:
-    runs-on: ubuntu-latest
+    runs-on: [self-hosted, ubuntu-latest]
```
Comment (bmwill) [inline .github/workflows/explorer-client-prs.yml:5]: This is a cheap job so maybe don't put it on the runners?

File: .github/workflows/explorer-client-prs.yml [hunk 4]
```diff
@@ -14,7 +14,7 @@ jobs:
     name: Lint, Test & Build
     needs: diff
     if: needs.diff.outputs.isClient == 'true'
-    runs-on: ubuntu-latest
+    runs-on: [self-hosted, ubuntu-latest]
```
Comment (bmwill) [inline .github/workflows/explorer-client-prs.yml:17]: This is a cheap job so maybe don't put it on the runners?

File: .github/workflows/rust.yml [hunk 5]
```diff
@@ -41,15 +41,15 @@ jobs:
 
   license-check:
     name: license-check
-    runs-on: ubuntu-latest
+    runs-on: [self-hosted, ubuntu-latest]
```
Comment (bmwill) [inline .github/workflows/rust.yml:44]: This is a cheap job so maybe don't put it on the runners?

File: .github/workflows/rust.yml [hunk 6]
```diff
@@ -30,7 +30,7 @@ env:
 
 jobs:
   diff:
-    runs-on: ubuntu-latest
+    runs-on: [self-hosted, ubuntu-latest]
```
Comment (bmwill) [inline .github/workflows/rust.yml:33]: This is a cheap job so maybe don't put it on the runners?

File: .github/workflows/rust.yml [hunk 7]
```diff
@@ -87,7 +87,7 @@ jobs:
   rustfmt:
     needs: diff
     if: needs.diff.outputs.isRust == 'true'
-    runs-on: ubuntu-latest
+    runs-on: [self-hosted, ubuntu-latest]
```
Comment (bmwill) [inline .github/workflows/rust.yml:90]: This is a cheap job so maybe don't put it on the runners?

File: .github/workflows/rust.yml [hunk 8]
```diff
@@ -103,15 +103,15 @@ jobs:
     name: cargo-deny (advisories, licenses, bans, ...)
     needs: diff
     if: needs.diff.outputs.isRust == 'true'
-    runs-on: ubuntu-latest
+    runs-on: [self-hosted, ubuntu-latest]
```
Comment (bmwill) [inline .github/workflows/rust.yml:106]: This is a cheap job so maybe don't put it on the runners?

File: .github/workflows/rust.yml [hunk 9]
```diff
@@ -103,15 +103,15 @@ jobs:
...
     steps:
     - uses: actions/checkout@v3
     - uses: EmbarkStudios/cargo-deny-action@v1
 
   cargo-udeps:
     needs: diff
     if: needs.diff.outputs.isRust == 'true'
-    runs-on: ubuntu-latest
+    runs-on: [self-hosted, ubuntu-latest]
```
Comment (bmwill) [inline .github/workflows/rust.yml:114]: This is a cheap job so maybe don't put it on the runners?

**Review by bmwill — APPROVED** [review]
marked some jobs we may not want on the self-hosted runners in order to prevent spending more $$ than we need to.

File: .github/workflows/rust.yml (hunk 9 above)
Comment (todd-mystenlabs) [reply .github/workflows/rust.yml:114]: Okay I may come back and lift that one out.

**Pull request merged**